temporal-spring-boot-autoconfigure-alpha/src/main/java/io/temporal/spring/boot/autoconfigure/WorkersAutoConfiguration.java

```diff
@@ -0,0 +1,244 @@
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.config.BeanPostProcessor;
+import org.springframework.beans.factory.config.ConfigurableListableBeanFactory;
+import org.springframework.beans.factory.support.BeanDefinitionValidationException;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.ApplicationListener;
+import org.springframework.context.annotation.*;
+import org.springframework.context.event.ContextRefreshedEvent;
+
+/** Provides a client based on `spring.temporal.workers` section */
+@Configuration
+@EnableConfigurationProperties(TemporalProperties.class)
+@ConditionalOnClass(name = "io.temporal.worker.WorkerFactory")
+@Conditional(WorkersPresentCondition.class)
+@Import({WorkersAutoConfiguration.ActivityExplicitConfigProcessor.class})
+public class WorkersAutoConfiguration {
+  private static final Logger log = LoggerFactory.getLogger(WorkersAutoConfiguration.class);
+
+  private final TemporalProperties properties;
+  private final ConfigurableListableBeanFactory beanFactory;
+
+  public WorkersAutoConfiguration(
+      TemporalProperties properties, ConfigurableListableBeanFactory beanFactory) {
+    this.properties = properties;
+    this.beanFactory = beanFactory;
+  }
+
+  @Bean(name = "temporalWorkerFactory", destroyMethod = "shutdown")
+  public WorkerFactory workerFactory(
+      WorkflowClient workflowClient,
+      @Qualifier("temporalTestWorkflowEnvironment") @Autowired(required = false) @Nullable
+          TestWorkflowEnvironment testWorkflowEnvironment) {
+    if (testWorkflowEnvironment != null) {
+      return testWorkflowEnvironment.getWorkerFactory();
+    } else {
+      return WorkerFactory.newInstance(workflowClient);
+    }
+  }
+
+  @Bean(name = "temporalWorkers")
+  public Collection<Worker> workers(
+      WorkerFactory workerFactory,
+      @Qualifier("autoDiscoveredWorkflowImplementations") @Autowired(required = false) @Nullable
+          Collection<Class<?>> autoDiscoveredWorkflowImplementationClasses) {
+    Set<Worker> workers =
+        properties.getWorkers().stream()
+            .map(workerProperties -> newWorker(workerFactory, workerProperties))
+            .collect(Collectors.toSet());
+
+    if (autoDiscoveredWorkflowImplementationClasses != null) {
+      for (Class<?> clazz : autoDiscoveredWorkflowImplementationClasses) {
+        WorkflowImpl annotation = clazz.getAnnotation(WorkflowImpl.class);
+        for (String taskQueue : annotation.taskQueues()) {
+          log.info(
+              "Registering auto-discovered workflow class {} on a task queue {}", clazz, taskQueue);
+          Worker worker = workerFactory.newWorker(taskQueue);
```

Comment:
> and this will create three workers which will cause an issue.
`newWorker` reuses workers. With warning in log though. It never was possible to create several workers on the same task queue on the same WorkerFactory.
This will be changed later. I didn't want to touch SDK code in this PR and right now there is no getWorker on WorkerFactory, while there should be,

Reply:
Not possible with the current API. I will be adding a new API for WorkerFactory that will allow doing that.

Reply:
I don't see a reason for WorkerFactory shouldn't expose helper methods that make it easier. Our users should have an easier time implementing code like that themselves if they want it. workers map is already inside WorkerFactory.

Reply:
Yeah, if we error on multiple configured workers of the same task queue, that's fine.